### PR TITLE
doc: Updates online archive resource docs

### PR DESCRIPTION
**Pull request by zach-carr**
## Description

https://jira.mongodb.org/browse/DOCSP-33592

For technical review: could you please take a close look at the example update and make sure it's accurate? The example that the ticket referenced 404s now. 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by corryroot — APPROVED** [review]
LGTM!

File: website/docs/r/online_archive.html.markdown [hunk 1]
```diff
@@ -88,7 +93,7 @@ resource "mongodbatlas_online_archive" "test" {
...
+* `partition_fields` - (Recommended) Fields to use to partition data. You can specify up to two frequently queried fields (or up to three fields when one of them is ``date_field``) to use for partitioning data. Queries that don’t contain the specified fields require a full collection scan of all archived documents, which takes longer and increases your costs. To learn more about how partition improves query performance, see [Data Structure in S3](https://docs.mongodb.com/datalake/admin/optimize-query-performance/#data-structure-in-s3). The value of a partition field can be up to a maximum of 700 characters. Documents with values exceeding 700 characters are not archived. See [partition fields](#partition).
```
Comment (marcosuma) [inline website/docs/r/online_archive.html.markdown]: Was it intentional to use the double \`\`?
```suggestion
* `partition_fields` - (Recommended) Fields to use to partition data. You can specify up to two frequently queried fields (or up to three fields when one of them is `date_field`) to use for partitioning data. Queries that don’t contain the specified fields require a full collection scan of all archived documents, which takes longer and increases your costs. To learn more about how partition improves query performance, see [Data Structure in S3](https://docs.mongodb.com/datalake/admin/optimize-query-performance/#data-structure-in-s3). The value of a partition field can be up to a maximum of 700 characters. Documents with values exceeding 700 characters are not archived. See [partition fields](#partition).
```

Comment (zach-carr) [reply website/docs/r/online_archive.html.markdown]: No it wasn't, I'm too used to RST 😅  good catch

File: website/docs/r/online_archive.html.markdown [hunk 2]
```diff
@@ -25,15 +25,20 @@ resource "mongodbatlas_online_archive" "test" {
...
     }
 
+    partition_fields {
+        field_name = "lastName"
+        order = 2
+    }
+
     criteria {
         type = "DATE"
         date_field = "created"
```
Comment (marcosuma) [inline website/docs/r/online_archive.html.markdown]: this field needs to be fixed. I run the terraform apply with this example and I got:

```
Error: error creating MongoDB Atlas Online Archive:: https://cloud-dev.mongodb.com/api/atlas/v2/groups/65f82c4dd4c91e6c61437623/clusters/cluster-test-3/onlineArchives POST: HTTP 400 Bad Request (Error code: "ONLINE_ARCHIVE_DATE_CRITERIA_NOT_IN_PARTITION") Detail: Date criteria is not specified in partition fields. Reason: Bad Request. Params: []
│
│   with mongodbatlas_online_archive.test,
│   on main.tf line 1, in resource "mongodbatlas_online_archive" "test":
│    1: resource "mongodbatlas_online_archive" "test" {
```
You should pick one of the three partition fields (e.g. date_field)

File: website/docs/r/online_archive.html.markdown [hunk 3]
```diff
@@ -25,15 +25,20 @@ resource "mongodbatlas_online_archive" "test" {
     db_name      = var.database_name
 
     partition_fields {
-        field_name = "firstName"
+        field_name = "date_field"
```
Comment (marcosuma) [inline website/docs/r/online_archive.html.markdown]:
```suggestion
        field_name = "dateField"
```

Comment (marcosuma) [reply website/docs/r/online_archive.html.markdown]: nit: suggest to use the same text format for all field names

**Comment by marcosuma** [comment]
@zach-carr gentle ping here if you have time to address my points we can merge this :) Thank you!

**Comment by zach-carr** [comment]
Hey @marcosuma, thanks for your comments and sorry for the delay! Made those changes, please take another look whenever you get a chance

File: website/docs/r/online_archive.html.markdown (hunk 2 above)
Comment (zach-carr) [reply website/docs/r/online_archive.html.markdown]: now "dateField"

**Review by oarbusi — APPROVED** [review]

**Pull request merged**